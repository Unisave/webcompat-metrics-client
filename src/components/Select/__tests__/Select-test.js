/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import React from "react";
import renderer from "react-test-renderer";
import sinon from "sinon";
import { shallow } from "enzyme";

import Component from "..";

it("renders Select default correctly", () => {
  const tree = renderer
    .create(
      <Component
        name="browserList"
        optionList={[
          { label: "All browsers" },
          { label: "Firefox", value: "firefox" },
          { label: "Chrome", value: "chrome" },
          { label: "Edge", value: "edge" },
          { label: "Safari", value: "safari" },
        ]}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders Select with defaultValue", () => {
  const tree = renderer
    .create(
      <Component
        name="browserList"
        value={"firefox"}
        optionList={[
          { label: "All browsers" },
          { label: "Firefox", value: "firefox" },
          { label: "Chrome", value: "chrome" },
          { label: "Edge", value: "edge" },
          { label: "Safari", value: "safari" },
        ]}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders Select required", () => {
  const tree = renderer
    .create(
      <Component
        name="browserList"
        value={"firefox"}
        required={true}
        optionList={[
          { label: "All browsers" },
          { label: "Firefox", value: "firefox" },
          { label: "Chrome", value: "chrome" },
          { label: "Edge", value: "edge" },
          { label: "Safari", value: "safari" },
        ]}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders Select with style injected", () => {
  const tree = renderer
    .create(
      <Component
        name="browserList"
        value={"firefox"}
        style={{ color: "red" }}
        optionList={[
          { label: "All browsers" },
          { label: "Firefox", value: "firefox" },
          { label: "Chrome", value: "chrome" },
          { label: "Edge", value: "edge" },
          { label: "Safari", value: "safari" },
        ]}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders Select: simulates onChange events", () => {
  const onChange = sinon.spy();
  const event = { target: { name: "browserList", value: "chrome" } };
  const wrapper = shallow(
    <Component
      name="browserList"
      value={"firefox"}
      optionList={[
        { label: "All browsers" },
        { label: "Firefox", value: "firefox" },
        { label: "Chrome", value: "chrome" },
        { label: "Edge", value: "edge" },
        { label: "Safari", value: "safari" },
      ]}
      onChange={onChange}
    />,
  );
  wrapper.find("select").simulate("change", event);
  expect(onChange.calledOnce).toEqual(true);
});
