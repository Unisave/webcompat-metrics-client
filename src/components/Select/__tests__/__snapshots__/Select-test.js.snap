// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Input with style injected 1`] = `
<div
  className="component"
>
  <select
    className="select"
    name="browserList"
    onChange={[Function]}
    style={
      Object {
        "color": "red",
      }
    }
    value="firefox"
  >
    <option
      value=""
    >
      All browsers
    </option>
    <option
      value="firefox"
    >
      Firefox
    </option>
    <option
      value="chrome"
    >
      Chrome
    </option>
    <option
      value="edge"
    >
      Edge
    </option>
    <option
      value="safari"
    >
      Safari
    </option>
  </select>
  <div
    aria-hidden="true"
    className="arrow"
  />
</div>
`;

exports[`renders Select default correctly 1`] = `
<div
  className="component"
>
  <select
    className="select"
    name="browserList"
    onChange={[Function]}
    value=""
  >
    <option
      value=""
    >
      All browsers
    </option>
    <option
      value="firefox"
    >
      Firefox
    </option>
    <option
      value="chrome"
    >
      Chrome
    </option>
    <option
      value="edge"
    >
      Edge
    </option>
    <option
      value="safari"
    >
      Safari
    </option>
  </select>
  <div
    aria-hidden="true"
    className="arrow"
  />
</div>
`;

exports[`renders Select required 1`] = `
<div
  className="component"
>
  <select
    className="select"
    name="browserList"
    onChange={[Function]}
    required={true}
    value="firefox"
  >
    <option
      value=""
    >
      All browsers
    </option>
    <option
      value="firefox"
    >
      Firefox
    </option>
    <option
      value="chrome"
    >
      Chrome
    </option>
    <option
      value="edge"
    >
      Edge
    </option>
    <option
      value="safari"
    >
      Safari
    </option>
  </select>
  <div
    aria-hidden="true"
    className="arrow"
  />
</div>
`;

exports[`renders Select with defaultValue 1`] = `
<div
  className="component"
>
  <select
    className="select"
    name="browserList"
    onChange={[Function]}
    value="firefox"
  >
    <option
      value=""
    >
      All browsers
    </option>
    <option
      value="firefox"
    >
      Firefox
    </option>
    <option
      value="chrome"
    >
      Chrome
    </option>
    <option
      value="edge"
    >
      Edge
    </option>
    <option
      value="safari"
    >
      Safari
    </option>
  </select>
  <div
    aria-hidden="true"
    className="arrow"
  />
</div>
`;
